bee-protocol/src/workers/message/processor.rs

```
@@ -0,0 +1,222 @@
+                                if let Err(e) =
+                                    unreferenced_inserted_worker.send(UnreferencedMessageInserterWorkerEvent(
+                                        message_id,
+                                        tangle.get_latest_milestone_index(),
+                                    ))
+                                {
+                                    error!("Sending message to unreferenced inserter failed: {}.", e);
+                                }
+                            }
+                        };
+
+                        if payload_worker
+                            .send(PayloadWorkerEvent {
+                                message_id,
+                                message: message.clone(),
+                            })
+                            .is_err()
+                        {
+                            error!("Sending message {} to payload worker failed.", message_id);
+                        }
+
+                        notify_message(message_id, notifier);
+
+                        bus.dispatch(MessageProcessed { message_id });
+
+                        // TODO: boolean values are false at this point in time? trigger event from another location?
+                        bus.dispatch(VertexCreated {
+                            message_id,
+                            parent_message_ids: message.parents().to_vec(),
+                            is_solid: false,
+                            is_referenced: false,
+                            is_conflicting: false,
+                            is_milestone: false,
+                            is_tip: false,
+                            is_selected: false,
+                        });
+
+                        metrics.new_messages_inc();
+                    }
+                });
+            }
+
+            while let Some(event) = receiver.next().await {
+                let _ = tx.send(event).await;
```

are we ignoring the error?